<?xml version="1.0"?>
<robot name="AL5D">
 
  <link name="base_link">
    <visual>
      <origin xyz="0 0 0.020" rpy="0 0 0" />
      <geometry>
    <cylinder radius="0.045" length="0.040" />
      </geometry>
      <material name="black">
    <color rgba="0.5 0.5 0.5 1" />
      </material>
    </visual>
  </link>
 
  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.04" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-2.61" upper="2.61" effort="0.5" velocity="6.0" />
  </joint>
 
  <link name="link1">
    <visual>
      <origin xyz="0 0 0.005" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.045" length="0.01" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 0 0.025" rpy="0 0 0" />
      <geometry>
        <box size="0.06 0.05 0.030" />
      </geometry>
      <material name="black" />
    </visual>
  </link>
 
  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0.015 0 0.027" rpy="0 0 0" />
    <axis xyz="0 1 0" />
    <limit lower="-1.75" upper="1.75" effort="0.5" velocity="6.0" />
  </joint>
 
  <link name="link2">
    <visual>
      <origin xyz="0 0 0" rpy="1.57079632675 0 0" />
      <geometry>
        <cylinder radius="0.013" length="0.06" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 0 0.0525" rpy="0 0 0" />
      <geometry>
        <box size="0.026 0.06 0.105" />
      </geometry>
      <material name="black" />
    </visual>
     <visual>
      <origin xyz="0 0 0.105" rpy="1.57079632675 0 0" />
      <geometry>
        <cylinder radius="0.013" length="0.06" />
      </geometry>
      <material name="black" />
    </visual>
  </link>
 
  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 0 0.105" rpy="0 0 0" />
    <axis xyz="0 -1 0" />
    <limit lower="-2.55" upper="2.55" effort="0.5" velocity="6.0" />
  </joint>
 
  <link name="link3">
    <visual>
      <origin xyz="0 0 -0.0125" rpy="0 0 0" />
      <geometry>
        <box size="0.026 0.05 0.06" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 0 0.06" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.0105" length="0.120" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 0 0.10" rpy="0 0 0" />
      <geometry>
        <box size="0.026 0.05 0.06" />
      </geometry>
      <material name="black" />
    </visual>
  </link>
 
  <joint name="joint4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz="0 0 0.120" rpy="0 0 0" />
    <axis xyz="0 -1 0" />
    <limit lower="-1.80" upper="1.80" effort="0.5" velocity="6.0" />
  </joint>
 
  <link name="link4">
    <visual>
      <origin xyz="0 0 0" rpy="1.57079632675 0 0" />
      <geometry>
        <cylinder radius="0.013" length="0.06" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 0 0.015" rpy="0 0 0" />
      <geometry>
        <box size="0.026 0.06 0.03" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 0.026 0.0375" rpy="0 0 0" />
      <geometry>
        <box size="0.013 0.013 0.0375" />
      </geometry>
      <material name="black" />
    </visual>
    <visual>
      <origin xyz="0 -0.026 0.0375" rpy="0 0 0" />
      <geometry>
        <box size="0.013 0.013 0.0375" />
      </geometry>
      <material name="black" />
    </visual>
  </link>
</robot>
